File: drestaurant-apps/drestaurant-monolith/src/main/java/com/drestaurant/query/repository/CustomerRepository.java
package com.drestaurant.query.repository;

import com.drestaurant.query.model.CustomerEntity;
import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;
import org.springframework.data.rest.core.annotation.RestResource;

@RepositoryRestResource(collectionResourceRel = "customers", path = "customers") public interface CustomerRepository
		extends PagingAndSortingRepository<CustomerEntity, String> {

	@Override @RestResource(exported = false) CustomerEntity save(CustomerEntity entity);

	@Override @RestResource(exported = false) void deleteById(String aLong);

	@Override @RestResource(exported = false) void delete(CustomerEntity entity);

}

File: drestaurant-apps/drestaurant-monolith-rest/src/main/kotlin/com/drestaurant/query/repository/CustomerRepository.kt
package com.drestaurant.query.repository

import com.drestaurant.query.model.CustomerEntity
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.data.rest.core.annotation.RepositoryRestResource
import org.springframework.data.rest.core.annotation.RestResource

@RepositoryRestResource(collectionResourceRel = "customers", path = "customers")
interface CustomerRepository : PagingAndSortingRepository<CustomerEntity, String> {

    @RestResource(exported = false)
    override fun deleteById(aLong: String)

    @RestResource(exported = false)
    override fun delete(entity: CustomerEntity)

    @RestResource(exported = false)
    override fun deleteAll(entities: MutableIterable<CustomerEntity>)

    @RestResource(exported = false)
    override fun <S : CustomerEntity?> save(entity: S): S

    @RestResource(exported = false)
    override fun <S : CustomerEntity?> saveAll(entities: MutableIterable<S>): MutableIterable<S>

    @RestResource(exported = false)
    override fun deleteAll()
}
